### ISSUE TITLE
Model computed_fields from a Model Type  

### ISSUE THREAD

**Issue by SylvainGuieu** [pydantic/pydantic#8573]
### Initial Checks

- [X] I confirm that I'm using Pydantic V2

### Description

Not sure if this is a bug or a feature. 
So far   (pydantic v2.5)   , the  model_computed_fields (a property) only work  on model instances  and not from model class. 

I do not see obvious reason for that.  See code below.  
A solution would be to not use builtin python `@property`  but a custom  property with a re-writen `__get__` method 

### Example Code

```Python
from pydantic import BaseModel, computed_field  

class M(BaseModel):
    x: float = 1.0 

    @computed_field
    @property 
    def tenx(self)->float:
       return self.x*10

>>> M.model_computed_fields
<property at 0x7fc4cbcf2860>
>>> M.model_computed_fields.__get__(None, M) # actually same as above 
<property at 0x7fc4cbcf2860>
>>> M.model_computed_fields.__get__(M, None) 
{'tenx': ComputedFieldInfo(wrapped_property=<property object at 0x7fc4cbc84360>, return_type=<class 'float'>, alias=None, alias_priority=None, title=None, description=None, examples=None, json_schema_extra=None, repr=True)}
```

### Python, Pydantic & OS Version

```Text
pydantic version: 2.5.3
        pydantic-core version: 2.14.6
          pydantic-core build: profile=release pgo=false
                 install path: /Users/guieus/anaconda3/envs/hrmpm/lib/python3.9/site-packages/pydantic
               python version: 3.9.18 (main, Sep 11 2023, 08:38:23)  [Clang 14.0.6 ]
                     platform: macOS-10.16-x86_64-i386-64bit
             related packages: typing_extensions-4.9.0
```

**Comment by SylvainGuieu** [comment]
I think one can actually add  a property to the BaseModel Metaclass to fix it. 
I did the following test, it does not seems to be a problem in python :
 
```python

class MMeta(type):
     @property
     def model_computed_fields(cls):
          return 1 

class M(metaclass=MMeta):
     @property 
     def model_computed_fields(self):
           return 2  

assert M.model_computed_fields == 1
assert M().model_computed_fields == 2
```

**Comment by sydney-runkle** [comment]
Fixed here:

https://github.com/pydantic/pydantic/pull/8437

Should be released with 2.6 soon
